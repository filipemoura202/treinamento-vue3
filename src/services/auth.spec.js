import mockAxios from 'axios'
import AuthService from './auth'

jest.mock('axios')

describe('Auth Service', () => {
  afterEach(() => {
    jest.clearAllMocks()
  })

  it('Should return a token user login', async () => {
    const token = '123.123.123'

    mockAxios.post.mockImplementationOnce(() => {
      return Promise.resolve({ data: { token } })
    })

    const response = await AuthService(mockAxios).login({ email: 'filipe@gmail.com', password: '1234' })
    expect(response.data).toHaveProperty('token')
    expect(response).toMatchSnapshot()
  })

  it('Should return an user when user register', async () => {
    const user = {
      name: 'Filipe',
      email: 'filipe@gmail.com',
      password: '1234'
    }

    mockAxios.post.mockImplementationOnce(() => {
      return Promise.resolve({ data: user })
    })

    const response = await AuthService(mockAxios).register(user)
    expect(response.data).toHaveProperty('name')
    expect(response.data).toHaveProperty('email')
    expect(response.data).toHaveProperty('password')
    expect(response).toMatchSnapshot()
  })

  it('Should throw an error when not Found', async () => {
    const errors = {
      status: 404,
      statusText: 'Not Found'
    }

    mockAxios.post.mockImplementationOnce(() => {
      return Promise.resolve({ request: errors })
    })

    const response = await AuthService(mockAxios).login({
      email: 'filipe@gmail.com', password: '1234'
    })
    expect(response.errors).toHaveProperty('status')
    expect(response.errors).toHaveProperty('statusText')
    expect(response).toMatchSnapshot()
  })
})
