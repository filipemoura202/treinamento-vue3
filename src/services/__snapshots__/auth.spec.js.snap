// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth Service Should return a token user login 1`] = `
Object {
  "data": Object {
    "token": "123.123.123",
  },
  "errors": null,
}
`;

exports[`Auth Service Should return an user when user register 1`] = `
Object {
  "data": Object {
    "email": "filipe@gmail.com",
    "name": "Filipe",
    "password": "1234",
  },
  "errors": null,
}
`;

exports[`Auth Service Should throw an error when not Found 1`] = `
Object {
  "data": undefined,
  "errors": Object {
    "status": 404,
    "statusText": "Not Found",
  },
}
`;
